pkg/assembler/graphql/schema/metadata.graphql

```diff
@@ -0,0 +1,44 @@
+#
+# Copyright 2023 The GUAC Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# NOTE: This is experimental and might change in the future!
+
+# Defines a GraphQL schema for the HasMetadata
+
+"""
+HasMetadata is an attestation that a package, source, or artifact has a certain
+attested property (key) with value (value). For example, a source may have
+metadata "SourceRepo2FAEnabled=true". 
+
+The intent of this evidence tree predicate is to allow extensibility of metadata
+expressible within the GUAC ontology. Metadata that is commonly used will then
+be promoted to a predicate on its own.
+
+Justification indicates how the metadata was determined.
+
+The metadata applies to a subject which is a package, source, or artifact.
+If the attestation targets a package, it must target a PackageName or a
+PackageVersion. If the attestation targets a source, it must target a
+SourceName.
+"""
+type HasMetadata {
+  id: ID!
+  subject: PackageSourceOrArtifact!
+  justification: String!
+  key: String!
```

Comment:
should this be an array of key-value pairs? similar to how we have `PackageQualifier`?

Reply:
i think it would be different nodes, and they would collapse to be evaluated since they could come from multiple sources.

Reply:
What if one source has a bunch of data? Just thinking if it will cause number is nodes to grow unnecessarily.

Reply:
We don't really have good support for arbitrary lists in nodes in the backend. In GraphQL we can make one single type that contains a list, but in the backend we will need to implement these separately

Reply:
Why not do something similar to package qualifiers and hassbom annotations. Seems like having multiple key value pairs per source makes more sense.

Reply:
We still cannot search for all packages that have a certain list of qualifiers and others (like search for "arch=some_value" but also get package that has "os=windows" as an additional qualifier)

Reply:
If that is the case, we should remove all key/value pair lists from graphQL?

Reply:
Here it's more likely to want to search for a single key, than in the package qualifier.

Reply:
hmm...we should remove annotations from hasSBOM. This can be used instead. I will make a PR for that, as that will also be something we would want to search for using a single key.

Reply:
created issue https://github.com/guacsec/guac/issues/1018